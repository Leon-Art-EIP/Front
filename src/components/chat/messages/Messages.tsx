import { useChat } from "../../../contexts/ChatContext";
import { ChatBox } from "./ChatBox";
import { ChatUserBanner } from "./ChatUserBanner";
import { MessageInput } from "./MessageInput";

export default function Messages(): JSX.Element {
  const { currentUser, currentChat, sendMessage } = useChat() || {};

  function getOtherUser() {
    if (currentChat?.UserOneId === currentUser?.user.id) {
      return currentChat?.UserTwoId;
    } else {
      return currentChat?.UserOneId;
    }
  }

  function handleSendMsg(msg: string) {
    var to = getOtherUser();
    if (to) {
      sendMessage(msg, to);
    }
  }

  return (
    <div className="flex flex-col h-full w-full">
      {currentChat && <ChatUserBanner currentChat={currentChat} currentUser={currentUser} />}
      <div className="flex flex-row w-full" style={{ height: "calc(100% - 6rem)" }}>
        <div className="flex flex-col h-full w-full">
          <ChatBox />
          <MessageInput handleSendMsg={handleSendMsg} />
        </div>
      </div>
    </div>
  );
}
