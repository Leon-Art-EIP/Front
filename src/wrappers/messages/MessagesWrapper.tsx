"use client";

import { useEffect, useState } from "react";
import Conversation from "../../components/messages/Conversation";
import UserList from "../../components/messages/userList/UserList";
import { IConversation, IConversations } from "../../interfaces/messages/conversations";
import { useWindowSize } from "../../hooks/useWindowSize";
import { useConversationService } from "../../hooks/messages/useConversationService";

export default function MessagesWrapper(): JSX.Element {
  const [showConversation, setShowConversation] = useState(false);
  const windowSize = useWindowSize();
  const conversationService = useConversationService();

  return (
    // TODO: have to be h-full
    <div className="flex flex-col lg:flex-row h-screen">
      <div className="w-full h-full lg:w-1/3 lg:min-w-[350px] lg:max-w-[500px]">
        {!showConversation && <UserList conversationService={conversationService} />}
      </div>
      {((windowSize.width && windowSize.width > 640) || showConversation) && (
        <div className="w-full">
          <Conversation conversationService={conversationService} />
        </div>
      )}
    </div>
  );
}
