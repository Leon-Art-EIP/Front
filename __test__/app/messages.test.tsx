import { render } from "@testing-library/react";
import React from "react";
import { expect, test } from "vitest";
import Messages from "../../src/app/messages/page";
import MessagesWrapper from "../../src/wrappers/messages/MessagesWrapper";

test("Messages page test", () => {
  const { container } = render(<MessagesWrapper />);

  expect(container).toMatchSnapshot();
});
