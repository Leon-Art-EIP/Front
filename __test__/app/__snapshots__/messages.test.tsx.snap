// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Messages page test 1`] = `
<div>
  <div
    class="flex flex-row page-content-non-scrollable"
  >
    <div
      class="lg:w-1/3 lg:min-w-[350px] lg:max-w-[500px]"
    >
      <div
        class="flex flex-col h-full bg-gray-100 shadow-[3px_0_3px_0px_rgba(170,170,170)]"
      >
        <div
          class="m-6 sm:m-10"
        >
          <div
            class="w-full"
          >
            <div
              class="relative flex items-center w-full h-12 rounded-3xl focus-within:shadow-lg bg-white overflow-hidden"
            >
              <input
                class="peer h-full w-full outline-none text-md placeholder-gray-600 pl-6"
                id="search"
                placeholder="Rechercher ..."
                type="text"
              />
              <div
                class="grid place-items-center h-full w-12 text-gray-300"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-1 gap-3"
        />
      </div>
    </div>
    <div
      class="flex flex-col h-full w-full"
    >
      <div
        class="flex flex-row h-24 p-4 items-center gap-4 shadow-[0px_6px_7px_0px_rgba(170,170,170)] z-10"
      >
        <span
          class="w-16 h-16 rounded-full bg-gray-400 flex-shrink-0"
        />
        <span
          class="text-2xl"
        />
      </div>
      <div
        class="flex flex-row w-full"
        style="height: calc(100% - 6rem);"
      >
        <div
          class="flex flex-col h-full w-full"
        >
          <div
            class="flex flex-col h-full w-full overflow-y-auto gap-4"
            dir="btt"
          />
          <div
            class="flex justify-center items-center h-24"
          >
            <form
              class="flex flex-row gap-4 justify-between bg-[#F3F3F3] w-full rounded-full mx-4 my-2"
            >
              <button
                class="bg-[#FFE3E3] p-3 rounded-full h-full text-center"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium h-7 w-7 rotate-45 fill-[#e11c0a] css-i4bv87-MuiSvgIcon-root"
                  data-testid="AttachmentIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2 12.5C2 9.46 4.46 7 7.5 7H18c2.21 0 4 1.79 4 4s-1.79 4-4 4H9.5C8.12 15 7 13.88 7 12.5S8.12 10 9.5 10H17v2H9.41c-.55 0-.55 1 0 1H18c1.1 0 2-.9 2-2s-.9-2-2-2H7.5C5.57 9 4 10.57 4 12.5S5.57 16 7.5 16H17v2H7.5C4.46 18 2 15.54 2 12.5z"
                  />
                </svg>
              </button>
              <input
                class="w-full bg-inherit outline-none placeholder:text-[#7d7d7d]"
                placeholder="Votre message..."
                value=""
              />
              <button
                class="bg-[#E3F3FF] p-3 rounded-full h-full text-center"
                type="submit"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium h-7 w-7 fill-[#6db1e3] css-i4bv87-MuiSvgIcon-root"
                  data-testid="SendIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                  />
                </svg>
              </button>
            </form>
          </div>
        </div>
        <div
          class="flex flex-col gap-4 hidden h-full p-4"
        />
      </div>
    </div>
  </div>
</div>
`;
